src/gasPriceOracle/adapters/polygon.ts

```diff
@@ -67,23 +67,28 @@ class PolygonGasStation extends BaseHTTPAdapter {
   }
 }
 
-export async function gasStation(provider: providers.Provider, chainId: number): Promise<GasPriceEstimate> {
+export async function gasStation(
+  provider: providers.Provider,
+  chainId: number,
+  markup: number
+): Promise<GasPriceEstimate> {
   const gasStation = new PolygonGasStation({ chainId: chainId, timeout: 2000, retries: 0 });
   let maxPriorityFeePerGas: BigNumber;
   let maxFeePerGas: BigNumber;
   try {
     ({ maxPriorityFeePerGas, maxFeePerGas } = await gasStation.getFeeData());
+    maxFeePerGas = maxFeePerGas.mul(markup);
   } catch (err) {
     // Fall back to the RPC provider. May be less accurate.
-    ({ maxPriorityFeePerGas, maxFeePerGas } = await eip1559(provider, chainId));
+    ({ maxPriorityFeePerGas, maxFeePerGas } = await eip1559(provider, chainId, markup));
 
     // Per the GasStation docs, the minimum priority fee on Polygon is 30 Gwei.
     // https://docs.polygon.technology/tools/gas/polygon-gas-station/#interpretation
     const minPriorityFee = parseUnits("30", 9);
     if (maxPriorityFeePerGas.lt(minPriorityFee)) {
       const priorityDelta = minPriorityFee.sub(maxPriorityFeePerGas);
       maxPriorityFeePerGas = minPriorityFee;
-      maxFeePerGas = maxFeePerGas.add(priorityDelta);
+      maxFeePerGas = maxFeePerGas.mul(markup).add(priorityDelta);
     }
   }
```

Comment:
For Polygon I think it makes sense to defer the markup until after this try/catch block, and to skip a markup on the call to `eip1559()`. Then it can be applied once, and doesn't risk double-padding (as with lines 83 + 91).

Reply:
won't this multiple the base fee a bit more in the case where the `catch` is hit and `maxPriorityFeePerGas.lt(minPriorityFee)`, then the markup gets applied to the maxFeePerGas + priorityDelta?

It's not really a big deal but is this truly a base fee markup or do we need more logic?

Reply:
4d76862ff39d4f8ad55a1a2c4d192a6a617f7976

Reply:
Hm; yeah you're right - I missed that. I think the change is OK and keeps the code simple, though if it was any chain other than Polygon it'd probably make sense to revert to your original implementation and set `eip1559(provider, chainId, 1.0)` instead.

Reply:
This might be more accurate actually: b4d4847989d5495442ffd6e92e2740032d01ec02

Reply:
Yep - looks good!